### PR TITLE
feat: document suffix replacements and multiple exclude files

### PR DESCRIPTION
**Pull request by theseion**

### CODE REVIEW

**Comment by RedXanadu** [comment]
@theseion Sorry, it's been a busy-ish week on my regular job… I've set aside some time on Monday to review this and all of the bits you've asked me to review. I haven't forgotten :smile:

**Comment by theseion** [comment]
No worries 🙂

File: content/development/regex_assembly.md [hunk 1]
```diff
@@ -313,14 +313,15 @@ Processor name: `include`
 ### Arguments
 
 - Include file name (required): The name of the file to include, without suffix
+- Suffix replacements (optional): Any number of space separated two-tuples, where the first entry is the suffix to match and the second entry is the replacement. To use, write `--` after the include file name. Tuples are space separated
```
Comment (RedXanadu) [inline content/development/regex_assembly.md:316]: > Any number of space separated two-tuples… Tuples are space separated

Is the repetition intentional? Maybe it is an essential point and needs repeating, but I wasn't sure if it was deliberately repeated when I read it.

File: content/development/regex_assembly.md [hunk 2]
```diff
@@ -313,14 +313,15 @@ Processor name: `include`
...
 
 ### Output
 
-The exact contents of the included file, including processor directives. The prefix and suffix markers are not allowed in included files.
+The exact contents of the included file, including processor directives, with suffixes replaced where apopriate. The prefix and suffix markers are not allowed in included files.
```
Comment (RedXanadu) [inline content/development/regex_assembly.md:320]:
```suggestion
The exact contents of the included file, including processor directives, with suffixes replaced where appropriate. The prefix and suffix markers are not allowed in included files.
```

Typo

File: content/development/regex_assembly.md [hunk 3]
```diff
@@ -356,22 +357,25 @@ it{{quotes}}s{{opt-lazy-wspace}}possible
...
 ## Include-Except processor
 
 Processor name: `include-except`
 
 ### Arguments
 
 - Include file name (required): The name of the file to include, without suffix
-- Exclude file name (required): The name of the file to consult for exclusions, without suffix
+- Exclude file names (required): One or more names of files to consult for exclusions, without suffix, space separated
+- Suffix replacements (optional): Any number of space separated two-tuples, where the first entry is the suffix to match and the second entry is the replacement. To use, end the list of exclude file names with `--`. Tuples are space separated
```
Comment (RedXanadu) [inline content/development/regex_assembly.md:370]: Same comment as above regarding repetition: might be an oversight or maybe it's intentional.

File: content/development/regex_assembly.md (hunk 1 above)
Comment (theseion) [reply content/development/regex_assembly.md:316]: Nope, thanks.

File: content/development/regex_assembly.md [hunk 4]
```diff
@@ -356,22 +357,25 @@ it{{quotes}}s{{opt-lazy-wspace}}possible
...
 
 - Include file name (required): The name of the file to include, without suffix
-- Exclude file name (required): The name of the file to consult for exclusions, without suffix
+- Exclude file names (required): One or more names of files to consult for exclusions, without suffix, space separated
+- Suffix replacements (optional): Any number of two-tuples, where the first entry is the suffix to match and the second entry is the replacement. To use, end the list of exclude file names with `--`. Tuples are space separated
 
 ### Output
 
-The contents of the included file as per the include processor, but with all matching lines from the exclude file removed.
+The contents of the included file as per the include processor, but with all matching lines from the exclude file removed. Suffixes will have been replaced as apropriate.
```
Comment (RedXanadu) [inline content/development/regex_assembly.md:374]:
```suggestion
The contents of the included file as per the include processor, but with all matching lines from the exclude file removed. Suffixes will have been replaced as appropriate.
```

Ah crumbs, sorry, I should have caught this earlier :disappointed:

File: content/development/regex_assembly.md [hunk 5]
```diff
@@ -393,4 +397,20 @@ The regular expression for a rule at paranoia level 1 would then be generated by
...
+##!> include-except command-list pl1-exclude-list -- @ [\s<>] ~ [^\s]
+```
+
+`""` is the special literal used to represent the empty string in suffix replacements. In order to replace a suffix with the empty string one would write, for example:
+
+```python
+##!> include-except command-list pl1-exclude-list -- @ "" ~ ""
+```
+
+Suffix replacement is performed _after_ all exclusions have been removed, which means that entries in exclude files must target the _vebatim_ contents of the include file, i.e., `some entry@`, not `some entry[\s<>]`
```
Comment (RedXanadu) [inline content/development/regex_assembly.md:414]:
```suggestion
Suffix replacement is performed _after_ all exclusions have been removed, which means that entries in exclude files must target the _verbatim_ contents of the include file, i.e., `some entry@`, not `some entry[\s<>]`
```

And another. I somehow completely missed this earlier!

**Review by RedXanadu — APPROVED** [review]
Awesome! Let's goo :grin:

**Comment by theseion** [comment]
Thanks for the review!

**Pull request merged**